tensorflow/python/ops/nn_impl.py

```diff
@@ -1298,14 +1298,8 @@ def fused_batch_norm(
   # prevent exception (see cudnn.h).
   min_epsilon = 1.001e-5
   epsilon = epsilon if epsilon > min_epsilon else min_epsilon
-  # TODO(reedwm): In a few weeks, switch to using the V2 version exclusively. We
-  # currently only use the V2 version for float16 inputs, which is not supported
-  # by the V1 version.
-  if x.dtype == dtypes.float16 or x.dtype == dtypes.bfloat16:
-    fused_batch_norm_func = gen_nn_ops.fused_batch_norm_v2
-  else:
-    fused_batch_norm_func = gen_nn_ops._fused_batch_norm  # pylint: disable=protected-access
-  y, batch_mean, batch_var, _, _ = fused_batch_norm_func(
+
+  y, batch_mean, batch_var, _, _, _ = gen_nn_ops.fused_batch_norm_v3(
```

Comment:
@alextp mentioned that this needs a forward compatibility story. I think we need to do something like:
```
# N is the current checked-in value of forward_compatible_version plus 21
if tf.compat.forward_compatible_version <= N:
  ...old code paths with _fused_batch_norm and fused_batch_norm_v2
else:
  ...new code path with fused_batch_norm_v3
```
and in the test case (if any) we can add `tf.compat.forward_compatible_version = N + 1` before running the test.

@alextp would you please confirm if this is sufficient? Thanks.

Reply:
The API I mentioned is in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/compat/compat.py

Reply:
Hi @aaroey , Can you articulate this a little more? I want to get some example of how to use this forward compatibility and search `forward_compatible_version` in this github, but I get nothing. Is this a new feature/style?
 
What is the `N` (the current checked-in value)? Is there any API that can provide you a `N`? And do you need to explicitly add `21` (And why?)?

Reply:
I do find some usage of `compat.forward_compatible(year=2018, month=10, day=30)`. Should we use this instead and set the date to be 2019, 5, 14?

For the tests, it seems if the compatibility check fails, we would skip the test. Is this what you refer to?
https://github.com/tensorflow/tensorflow/blob/a6003151399ba48d855681ec8e736387960ef06e/tensorflow/python/kernel_tests/cwise_ops_test.py#L1120

Reply:
I used old APIs in the code snippet above, sorry for the confusion. It should be something like:
```
if compat.forward_compatible(2019, 6, 6):  # 21 days from when this PR is expected to be merged
  y, batch_mean, batch_var, _, _, _ = gen_nn_ops.fused_batch_norm_v3(...)
else:  # Repeat the old code
  if x.dtype == dtypes.float16 or x.dtype == dtypes.bfloat16:	
    fused_batch_norm_func = gen_nn_ops.fused_batch_norm_v2	
  else:	
    fused_batch_norm_func = gen_nn_ops._fused_batch_norm 
  y, batch_mean, batch_var, _, _ = fused_batch_norm_func(...)
```

For all the related python tests we need to do:
```
with compat.forward_compatibility_horizon(2019, 6, 7):  # Some time after the expiration of the compatibility window
  ...test that fused_batch_norm_v3 has an effect.
```

You may refer to the compat.py for the details. Thanks.